### PR TITLE
Use an options object for GCS credentials

### PR DESCRIPTION
**Pull request by adshmh**
Part of work on #13084

### CODE REVIEW

**Comment by adshmh** [comment]
@stevekuznetsov I limited the scope to the credentials file for the first draft, but it may be possible to factor out all `statusURI`, `historyURI`, etc. used by controllers (`tide` uses 2, other controllers use only 1).

File: prow/flagutil/storage.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	"flag"
+	"fmt"
+
+	"k8s.io/test-infra/pkg/io"
+)
+
+type StorageClientOptions struct {
+	// The following are used for reading/writing to GCS.
+	GCSCredentialsFile   string
+	GCSOptionDescription string
```
Comment (stevekuznetsov) [inline prow/flagutil/storage.go]: nit: why is this public, or stored? couldn't we store a closure to bind onto a flagset or at least store the description as a private member?

File: prow/flagutil/storage.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+// StorageClient returns a Storage client.
+func (o *StorageClientOptions) StorageClient(ctx context.Context) (io.Opener, error) {
+	opener, err := io.NewOpener(ctx, o.GCSCredentialsFile)
+	if err != nil {
+		gcsMessage := ""
+		if o.GCSCredentialsFile != "" {
+			gcsMessage = fmt.Sprintf(" gcs-credentials-file: %s", o.GCSCredentialsFile)
+		}
+		return opener, fmt.Errorf("error creating opener%s: %w", gcsMessage, err)
```
Comment (stevekuznetsov) [inline prow/flagutil/storage.go]: `s/%w/%v`?

Comment (adshmh) [reply prow/flagutil/storage.go]: Thanks for the review. Done.

File: prow/flagutil/storage.go (hunk 1 above)
Comment (adshmh) [reply prow/flagutil/storage.go]: Thank you for the review. Fixed.

**Comment by mikesplain** [comment]
I was just looking into doing this, anyway this can be re-reviewed @stevekuznetsov? Thanks!

/ok-to-test

**Pull request merged**